sdk/core/azure-core/azure/core/pipeline/_base.py

```diff
@@ -98,6 +98,7 @@ def send(self, request):
         :return: The PipelineResponse object.
         :rtype: ~azure.core.pipeline.PipelineResponse
         """
+        request.context.options.pop('insecure_domain_change', False)
```

Comment:
It seems a bit hacky to pop this here. The transport runner shouldn't clean up after pipeline policies that are leaking options.

Reply:
Agreed. But I could not find a good place to guarantee that it is popped.

You know we were working on a general clean up in transport (https://github.com/Azure/azure-sdk-for-python/pull/11284/files)

We can remove the hacky code after that one is ready.